server/src/main/java/org/elasticsearch/transport/TransportConnectionListener.java

```diff
@@ -38,5 +39,5 @@ default void onNodeConnected(DiscoveryNode node, Transport.Connection connection
     /**
      * Called once a node connection is closed and unregistered.
      */
-    default void onNodeDisconnected(DiscoveryNode node, Transport.Connection connection) {}
+    default void onNodeDisconnected(DiscoveryNode node, @Nullable Exception closeException) {}
```

Comment:
I [mentioned earlier](https://github.com/elastic/elasticsearch/pull/128415#discussion_r2117382024) that we _could_ consider pulling this API change out to a separate PR. As things stand I now think we should definitely do that - it's a simple refactoring (needs no test changes) and will make this change much more focussed.